packages/e2e-tests/specs/a11y.test.js

```diff
@@ -29,6 +30,36 @@ describe( 'a11y', () => {
 		expect( isFocusedToggle ).toBe( true );
 	} );
 
+	it( 'checks persistent selection', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'Testing editor selection persistence' );
+
+		await page.keyboard.down( 'Control' );
+		await page.keyboard.down( 'Shift' );
+		await page.keyboard.press( '`' );
+		await page.keyboard.press( '`' );
+		await page.keyboard.up( 'Control' );
+		await page.keyboard.up( 'Shift' );
+
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Space' );
```

Comment:
It might make it easier to follow this test if we add assertion whether the currently focused element is the Document tab button here and the Block tab button in the next step.